<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>661</width>
    <height>803</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>661</width>
    <height>803</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>661</width>
    <height>803</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>GBAATM-Rebirth</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="verticalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>9</x>
      <y>9</y>
      <width>641</width>
      <height>791</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <property name="sizeConstraint">
      <enum>QLayout::SetMinimumSize</enum>
     </property>
     <item>
      <widget class="QTabWidget" name="tabWidget">
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>360</height>
        </size>
       </property>
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="QWidget" name="tab_4">
        <attribute name="title">
         <string>Files</string>
        </attribute>
        <widget class="QWidget" name="formLayoutWidget">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>10</y>
           <width>611</width>
           <height>76</height>
          </rect>
         </property>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="1" column="1">
           <widget class="QLineEdit" name="output_path"/>
          </item>
          <item row="1" column="2">
           <widget class="QPushButton" name="output_btn">
            <property name="text">
             <string>Open</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Output</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QPushButton" name="input_btn">
            <property name="text">
             <string>Open</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="input_path"/>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Input</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QLabel" name="romname">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>100</y>
           <width>611</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
        <widget class="QLabel" name="checksum">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>130</y>
           <width>611</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
        <widget class="QLabel" name="warning">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>220</y>
           <width>611</width>
           <height>91</height>
          </rect>
         </property>
         <property name="mouseTracking">
          <bool>true</bool>
         </property>
         <property name="text">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;On GBATemp's threads (check &lt;span style=&quot; text-decoration: underline;&quot;&gt;Info&lt;/span&gt; tab) for this app and the original one &lt;span style=&quot; font-weight:600;&quot;&gt;there are tons of reports&lt;/span&gt;&lt;br/&gt;about some cheat codes that don't work or crashes the app.&lt;/p&gt;&lt;p&gt;The &lt;span style=&quot; text-decoration: underline;&quot;&gt;only&lt;/span&gt; solution is to try others or that includes the decryption key.&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600; text-decoration: underline;&quot;&gt;Or that someone fix the code&lt;/span&gt;.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </widget>
       </widget>
       <widget class="QWidget" name="tab_2">
        <attribute name="title">
         <string>CheatCodes</string>
        </attribute>
        <widget class="QPlainTextEdit" name="cheats">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>10</y>
           <width>611</width>
           <height>261</height>
          </rect>
         </property>
        </widget>
        <widget class="QPushButton" name="opencht">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>280</y>
           <width>101</width>
           <height>34</height>
          </rect>
         </property>
         <property name="text">
          <string>Open CHT file</string>
         </property>
        </widget>
        <widget class="QComboBox" name="mode">
         <property name="geometry">
          <rect>
           <x>120</x>
           <y>280</y>
           <width>181</width>
           <height>32</height>
          </rect>
         </property>
         <item>
          <property name="text">
           <string>Codebreaker/GS V3</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>RAW (VBA)</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>GS/AR v1/2 (Experimental)</string>
          </property>
         </item>
        </widget>
       </widget>
       <widget class="QWidget" name="tab_3">
        <attribute name="title">
         <string>Parameters</string>
        </attribute>
        <widget class="QWidget" name="gridLayoutWidget_4">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>50</y>
           <width>611</width>
           <height>41</height>
          </rect>
         </property>
         <layout class="QGridLayout" name="gridLayout_5">
          <item row="0" column="1">
           <widget class="QPushButton" name="loadselection">
            <property name="text">
             <string>Load selection bar - 240x14</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QPushButton" name="loadfont">
            <property name="text">
             <string>Load font texture - 9x12</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QPushButton" name="loadbg">
            <property name="text">
             <string>Load BG - 240x160</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QLineEdit" name="menu_text">
         <property name="geometry">
          <rect>
           <x>280</x>
           <y>10</y>
           <width>339</width>
           <height>32</height>
          </rect>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
        <widget class="QLabel" name="label_2">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>10</y>
           <width>191</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>Menu Text (use / to split lines):</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_8">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>110</y>
           <width>101</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>Execute every:</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_10">
         <property name="geometry">
          <rect>
           <x>360</x>
           <y>110</y>
           <width>81</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>RAM to use:</string>
         </property>
        </widget>
        <widget class="QCheckBox" name="vblank">
         <property name="geometry">
          <rect>
           <x>190</x>
           <y>110</y>
           <width>121</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>VBLANK Check</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
        <widget class="QSpinBox" name="execute_every">
         <property name="geometry">
          <rect>
           <x>100</x>
           <y>110</y>
           <width>52</width>
           <height>32</height>
          </rect>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>256</number>
         </property>
        </widget>
        <widget class="QComboBox" name="ram_block">
         <property name="geometry">
          <rect>
           <x>450</x>
           <y>110</y>
           <width>171</width>
           <height>32</height>
          </rect>
         </property>
         <item>
          <property name="text">
           <string>0x3007FA0</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>0x3007FC0</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>0x3007FD0</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>0x3007FE0</string>
          </property>
         </item>
        </widget>
        <widget class="QGroupBox" name="groupBox">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>220</y>
           <width>631</width>
           <height>101</height>
          </rect>
         </property>
         <property name="title">
          <string>Trainer</string>
         </property>
         <widget class="QWidget" name="gridLayoutWidget_3">
          <property name="geometry">
           <rect>
            <x>9</x>
            <y>30</y>
            <width>611</width>
            <height>61</height>
           </rect>
          </property>
          <layout class="QGridLayout" name="gridLayout_4">
           <item row="0" column="0">
            <widget class="QLabel" name="label_7">
             <property name="text">
              <string>Enable Keys:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLineEdit" name="trainer_enable_keys">
             <property name="text">
              <string>L+R+A</string>
             </property>
             <property name="placeholderText">
              <string/>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLabel" name="label_6">
             <property name="text">
              <string>Disable Keys:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLineEdit" name="trainer_disable_keys">
             <property name="text">
              <string>L+R+B</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </widget>
       <widget class="QWidget" name="tab_5">
        <attribute name="title">
         <string>Info</string>
        </attribute>
        <widget class="QTextBrowser" name="textBrowser">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>10</y>
           <width>611</width>
           <height>301</height>
          </rect>
         </property>
         <property name="html">
          <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Oxygen'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;Software based on GBAATM 1.9.9.7.0.0.1, made by Cracker.&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-size:10pt;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;On 2020 a new version after the release of the source code was possible. This new version is built with QT, include a refactoring of the original code that was ported to new operative systems with a new UI and &amp;quot;maybe&amp;quot; new features.&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-size:10pt;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;Code released, open for contributions: &lt;/span&gt;&lt;a href=&quot;https://github.com/Mte90/GBAATM-Rebirth&quot;&gt;&lt;span style=&quot; font-size:10pt; text-decoration: underline; color:#2980b9;&quot;&gt;https://github.com/Mte90/GBAATM-Rebirth&lt;/span&gt;&lt;/a&gt;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-size:10pt; text-decoration: underline; color:#2980b9;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Original app Thread: &lt;a href=&quot;https://github.com/Mte90/GBAATM-Rebirth&quot;&gt;&lt;span style=&quot; font-size:10pt; text-decoration: underline; color:#2980b9;&quot;&gt;https://gbatemp.net/threads/gba-auto-trainer-maker-gbaatm.99334&lt;/span&gt;&lt;/a&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;New app thread: &lt;a href=&quot;https://github.com/Mte90/GBAATM-Rebirth&quot;&gt;&lt;span style=&quot; font-size:10pt; text-decoration: underline; color:#2980b9;&quot;&gt;https://gbatemp.net/threads/gbaatm-rebirth-gba-auto-trainer-maker-a-new-hope.564321&lt;/span&gt;&lt;/a&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </widget>
       </widget>
      </widget>
     </item>
     <item>
      <layout class="QGridLayout" name="gridLayout">
       <property name="sizeConstraint">
        <enum>QLayout::SetMaximumSize</enum>
       </property>
       <property name="horizontalSpacing">
        <number>6</number>
       </property>
       <item row="0" column="1">
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="0" column="2">
        <widget class="QPushButton" name="patchgame">
         <property name="text">
          <string>Patch Game!</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QPushButton" name="deadbeef">
         <property name="text">
          <string>DEADBEEF</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QPlainTextEdit" name="log">
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>350</height>
        </size>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
